pandas/tests/generic/methods/test_dot.py

```diff
@@ -0,0 +1,128 @@
+import numpy as np
+import pytest
+
+from pandas import DataFrame, Series
+import pandas._testing as tm
+
+
+class DotSharedTests:
+    @pytest.fixture
```

Comment:
can you make these module level fixtures, having in a class is an anti-patter

Reply:
even better is to use the globally defined fixtures (but if hard nbd)

Reply:
@jbrockmendel do we need these class level fixtures like this?

Reply:
for some value of "need".  Simon and i do not have a consensus on this

Reply:
ok, the reason i question this is that we do not do this anywhere else (except for a PR you just did), and we DO this also in the resample tests. Since we aren't intentionally doing this globally I want to be sure this pattern is a valid / reasonable usecase that simply doesn't fit into our other patterns.

Reply:
We use it a bunch in tests.indexes